### ISSUE TITLE
migration-artifactory init container template generation is broken in artifactory-statefulset if resource requests are defined

### ISSUE THREAD

**Issue by ScOut3R** [jfrog/charts#1562]
**Version of Helm and Kubernetes**: 3.7.2 and 1.25.5

**Which chart**: artifactory

**What happened**:
```
Error: YAML parse error on artifactory-jcr/charts/artifactory/templates/artifactory-statefulset.yaml: error converting YAML to JSON: yaml: line 203: did not find expected key
```

**What you expected to happen**:
Template renders without issues

**How to reproduce it** (as minimally and precisely as possible):
Set resource requests for Artifactory.

**Anything else we need to know**:
The following highlight shows where the issue is. The `resources` inclusion breaks the list of volume mounts as after the resources we try to include any custom volume mounts.

https://github.com/jfrog/charts/blob/28ec91b8de8ee04ec486a9549d51a0843ef8e587/stable/artifactory/templates/artifactory-statefulset.yaml#L402-L405

The breaking change was in https://github.com/jfrog/charts/commit/28ec91b8de8ee04ec486a9549d51a0843ef8e587#diff-ea26e8e261a285b525d4f507037dfbabfaed4d29d51907febf8cff18ad433e5b

A rendered template with this change looks like this (snippet):
```yaml
volumeMounts:
        - name: migration-scripts
          mountPath: "/tmp/migrate.sh"
          subPath: migrate.sh
        - name: migration-scripts
          mountPath: "/tmp/migrationHelmInfo.yaml"
          subPath: migrationHelmInfo.yaml
        - name: migration-scripts
          mountPath: "/tmp/migrationStatus.sh"
          subPath: migrationStatus.sh
        - name: artifactory-volume
          mountPath: "/var/opt/jfrog/artifactory"
        - name: binarystore-xml
          mountPath: "/artifactory_bootstrap/binarystore.xml"
          subPath: binarystore.xml
        resources:
          limits:
            cpu: "3"
            memory: 14Gi
          requests:
            cpu: "3"
            memory: 14Gi
        - name: psql-ca-cert
          mountPath: "/home/artifactory/.postgresql/"
          readOnly: true
        - name: psql-ca-cert
          mountPath: "/etc/ssl/certs"
          readOnly: true
```

**Comment by chukka** [comment]
@ScOut3R Looks like bug , will try to fix this early and will plan to release this in coming release (ETA end of quarter release)

**Comment by chukka** [comment]
Fixed internally and will be part of 7.31.x release ( mostly 1st week of jan)

**Comment by ScOut3R** [comment]
> Fixed internally and will be part of 7.31.x release ( mostly 1st week of jan)

Thank you very much @chukka! I really appreciate the support. :medal_sports:

**Comment by chukka** [comment]
Fixed in #1573
